c3po/job/utils/metadata.py

```diff
@@ -1,38 +1,58 @@
-from c3po.db.dao import *
 from c3po.db.common.base import session_scope
+from c3po.db.dao import Artist, ArtistGenre, ArtistSong
+from c3po.db.dao import Genre
+from c3po.db.dao import Link
+from c3po.db.dao import Song, SongGenre
+from c3po.db.dao import User, UserLikes, UserPosts
 from datetime import datetime
+from isodate import parse_datetime, ISO8601Error
 from music_metadata_extractor import SongData
 
-def insert(url, user, date_time=datetime.now()):
+
+def insert_metadata(data):
+    url = data["link"]
     with session_scope() as session:
         data = SongData(url)
-        new_link = _insert_post(url, user, date_time, session)
-        if(new_link):
+        # This is a placeholder for until we can fetch real user details
+        user = _insert_default_user(session)
+        new_link = _insert_post(url, user, data, session)
```

Comment:
The data being passed to the `_insert_post()` method is a SongData object, whereas a JSON response from MongoDB is to be passed. Maybe we can have two variables: `raw_data` and `data`?

Reply:
Oh ow. Yes, that's definitely needed! Didn't notice the variable re-use.